api/src/main/java/com/capitalone/dashboard/config/WebSecurityConfig.java

```diff
@@ -0,0 +1,90 @@
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import com.capitalone.dashboard.auth.AuthProperties;
+import com.capitalone.dashboard.auth.AuthenticationResultHandler;
+import com.capitalone.dashboard.auth.ldap.LdapLoginRequestFilter;
+import com.capitalone.dashboard.auth.standard.StandardLoginRequestFilter;
+import com.capitalone.dashboard.auth.token.JwtAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+@EnableConfigurationProperties
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+	
+	@Autowired
+	private JwtAuthenticationFilter jwtAuthenticationFilter;
+	
+	@Autowired
+	private AuthenticationResultHandler authenticationResultHandler;
+
+	@Autowired
+	private AuthenticationProvider standardAuthenticationProvider;
+	
+	@Autowired
+	private AuthProperties authProperties;
+	
+	@Override
+	protected void configure(HttpSecurity http) throws Exception {
+		http.headers().cacheControl();
+		http.csrf().disable()
+			.authorizeRequests().antMatchers("/appinfo").permitAll()
+								.antMatchers("/registerUser").permitAll()
+								.antMatchers("/login**").permitAll()
+								.antMatchers(HttpMethod.GET, "/**").permitAll()
+								
+								// Temporary solution to allow jenkins plugin to send data to the api
+							    //TODO: Secure with API Key
+								.antMatchers(HttpMethod.POST, "/build").permitAll()
+					            .antMatchers(HttpMethod.POST, "/deploy").permitAll()
+					            .antMatchers(HttpMethod.POST, "/artifact").permitAll()
+					            .antMatchers(HttpMethod.POST, "/quality/test").permitAll()
+					            .antMatchers(HttpMethod.POST, "/quality/static-analysis").permitAll()
+					            
+								.anyRequest().authenticated()
+									.and()
+								.addFilterBefore(standardLoginRequestFilter(), UsernamePasswordAuthenticationFilter.class)
+								.addFilterBefore(ldapLoginRequestFilter(), UsernamePasswordAuthenticationFilter.class)
+								.addFilterBefore(jwtAuthenticationFilter, UsernamePasswordAuthenticationFilter.class)
+								.exceptionHandling().authenticationEntryPoint(new Http401AuthenticationEntryPoint("Authorization"));
+	}
+	
+    @Override
+	protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+		auth.authenticationProvider(standardAuthenticationProvider);
+		
+		String ldapServerUrl = authProperties.getLdapServerUrl();
+		String ldapUserDnPattern = authProperties.getLdapUserDnPattern();
+		if (StringUtils.isNotBlank(ldapServerUrl) && StringUtils.isNotBlank(ldapUserDnPattern)) {
+			auth.ldapAuthentication()
+			.userDnPatterns(ldapUserDnPattern)
+			.contextSource().url(ldapServerUrl);
```

Comment:
I might be interpreting this incorrectly, but I don't see any reason to retrieve the authorities at all with this implementation.  All the role specific stuff, such as who is admin, is stored in the Hygieia database.  This would give the tool the ability to add new admins, instead of relying on ldap groups and such.